### ISSUE TITLE
Feature Request: type in name instead of phone number

### ISSUE THREAD

**Issue by jdsmf** [mgolden/kdecsms#1]
[This fellow](https://doronbehar.com/articles/using-kdeconnect-to-comfortably-send-sms-messages-from-the-shell/) did something pretty cool using kdeconnect-cli, which utitlizes fzf for fuzzy searching through his list of contacts.  I wonder if something similar could be done with your nice qt gui.

**Comment by mgolden** [comment]
You can use my GUI with essentially the same script that he has. Instead of calling the command

kdeconnect-cli --device ${_KDECONNECT_DEFAULT_DEVICE} --send-sms "${args}" --destination "${phone_number}"

you would just replace that with

kdecsms.py "${phone_number}"

and then it would open the application and you can send your messages.

As described, there are a few shortcomings with Doron Behar's approach. (1) You have to manually look up the device ID and set into the environment variable. (2) He doesn't handle the case where the message doesn't send. This actually happens pretty frequently (at least for me) because Android can put the kdeconnect daemon on the phone to sleep and the phone disconnects. If you don't tell the user that this had happened the user will think that message went when it didn't. (3) You have to use khard and vidisyncer and set up a local address book.  The solution of using kdecsms instead of the kdeconnect-cli solves (1) and (2), but you still need to do (3). I think just using Thunderbird for that is better, but he may have a different use case. In Thunderbird you get the search-as-you-type stuff directly once you go into Cardbook.

What might be interesting would be to integrate kdecsms directly with vidisyncer so that you could look up people directly in the kdecsms gui. That would be useful for people who don't want to run Thunderbird Cardbook to keep their contacts.

**Comment by jdsmf** [comment]
Yeah, your last point is closer to what I was thinking, but maybe easier at first.  You could still integrate kdecsms with TB Cardbook, but be able to type names in the GUI that it would look up in TB CB.  Once that works, it might be easier to integrate with any other database. I like TB CB, but don't always want to keep it open.

**Comment by mgolden** [comment]
I did try to do that integration, actually. I couldn't find out any way to open the database from outside. I think that is a feature, not a bug, in the sense that the goal is to keep the contact data secure.

I may have missed something, so if someone has any knowledge as to how to do it I definitely would, because it's pretty much what I wanted to begin with.
